src/data/finsupp/basic.lean

```diff
@@ -632,6 +632,10 @@ def prod [has_zero M] [comm_monoid N] (f : α →₀ M) (g : α → M → N) : N
 
 variables [has_zero M] [has_zero M'] [comm_monoid N]
 
+@[to_additive]
+lemma prod_def (f : α →₀ M) (g : α → M → N) :
```

Comment:
There's no need for this lemma, you can use `rw prod` anywhere `rw prod_def` would work.